This fixes configure tests that mistakenly omit the -shared flag when
testing whether -znodelete can be used.

Need to file upstream bug

--- glib-2.44.0/configure.ac	2015-04-19 19:15:34.588825144 -0700
+++ glib-2.44.0/configure.ac	2015-04-19 19:18:29.068222772 -0700
@@ -3467,8 +3467,9 @@
 AC_ARG_ENABLE([znodelete],
               [AS_HELP_STRING([--disable-znodelete],
                               [avoid linking with -z,nodelete])],,
-              [SAVED_LDFLAGS="${LDFLAGS}" SAVED_LIBS="${LIBS}"
+              [SAVED_CFLAGS="${CFLAGS}" SAVED_LDFLAGS="${LDFLAGS}" SAVED_LIBS="${LIBS}"
                AC_MSG_CHECKING([for --fatal-warnings linker flag])
+               CFLAGS="${SAVED_CFLAGS} -shared"
                LDFLAGS=-Wl,--fatal-warnings
                LIBS=
                AC_TRY_LINK([], [return 0],
@@ -3483,7 +3484,7 @@
                            enable_znodelete=yes,
                            AC_MSG_RESULT(no)
                            enable_znodelete=no)
-               LDFLAGS="${SAVED_LDFLAGS}" LIBS="${SAVED_LIBS}"])
+               CFLAGS="${SAVED_CFLAGS}" LDFLAGS="${SAVED_LDFLAGS}" LIBS="${SAVED_LIBS}"])
 
 if test "x${enable_znodelete}" = "xyes"; then
   GOBJECT_LINK_FLAGS=-Wl,-z,nodelete
